src/components/AdvanceAnalyticsV2/tabs/Skills.jsx

```diff
@@ -1,10 +1,45 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useIntl } from '@edx/frontend-platform/i18n';
 import Header from '../Header';
-import EmptyChart from '../charts/EmptyChart';
+import BarChart from '../charts/BarChart';
+import ScatterChart from '../charts/ScatterChart';
+import ProgressOverlay from '../ProgressOverlay';
+import { useEnterpriseSkillsAnalytics } from '../data/hooks';
 
-const Skills = () => {
+
+const Skills = ({ enterpriseCustomerUUID, startDate, endDate }) => {
   const intl = useIntl();
+  console.log('startDate:', startDate);
+  console.log('endDate:', endDate);
+
```

Comment:
Note: I have tried to invalidate queries like below but it is not working. So I remove the `invalidateQueries` and include the startDate and endDate in useEnterpriseSkillsAnalytics but now it cache for every variation of enterpriseCustomerUUID, startDate, endDate which I am not sure is correct?

```javascript
useEffect(() => {
  queryClient.invalidateQueries({
    queryKey: advanceAnalyticsQueryKeys.skills(enterpriseCustomerUUID),
   });
}, [enterpriseCustomerUUID, startDate, endDate]);
```

Reply:
In the below screenshot of react-query dev tools, the cache time is set to 300 seconds which I am not sure why? Because I set the `staleTime` to `10800000` which is 6 hours. I am not sure what is happening here?

<img width="1435" alt="Screenshot 2024-08-21 at 6 37 31 PM" src="https://github.com/user-attachments/assets/93890bfa-4008-4c17-8655-d5c86e5bdd8a">

Reply:
Keep in mind, `staleTime` != `cacheTime`. See my related comment regarding how `cacheTime` should be larger than `staleTime`. If you're updating `staleTime` to 6 hours, the default `cacheTime` of 5 minutes means data is removed from the query cache after 5 minutes, even if the `staleTime` hasn't yet elapsed.

[aside] Are you sure `10800000` == 6 hours?

<img width="492" alt="image" src="https://github.com/user-attachments/assets/bce30dd8-fade-4c7c-bb31-7404fc8f9b54">

Reply:
> I have tried to invalidate queries like below but it is not working.

[inform] `invalidateQueries` is essentially telling React Query to "fetch the `queryFn` (API) again. It's usually used when needing to refetch data after some user mutation that changes the expected server API data.

Because your query key contains customer uuid and the start/end dates, if _any_ of those values change, a new fetch with the changed option will happen automatically and get stored in the query cache under the new query key. There's minimal concerns around leaving any previously cached data for the original options in the query cache as it'll eventually be garbage collected per the configured `cacheTime`.

Reply:
Apologies for my bad math 🤦‍♂️